Gems/Atom/RHI/Code/Include/Atom/RHI.Reflect/Limits.h

```diff
@@ -90,5 +92,7 @@ namespace AZ
         {
             constexpr int DefaultDeviceIndex = 0;
         }
+
+        constexpr uint32_t InvalidIndex = aznumeric_cast<uint32_t>(-1);
```

Comment:
I think using numeric_limits is better that casting from signed to unsigned here
```suggestion
        constexpr uint32_t InvalidIndex = AZStd::numeric_limits<uint32_t>::max;
```